llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp

```diff
@@ -439,6 +439,26 @@ struct AAAMDAttributesFunction : public AAAMDAttributes {
       indicatePessimisticFixpoint();
       return;
     }
+
+    SmallPtrSet<const Constant *, 8> VisitedConsts;
+
+    for (Instruction &I : instructions(F)) {
```

Comment:
I don't think it is generally a good idea to put this logic into `initialize` unless it can directly changes the status of the AA.